src/main/resources/assets/carpet/scripts/distance_rc_1.sc

```diff
@@ -0,0 +1,88 @@
+import('math','_euclidean','_manhattan','_round');
+
+__config()->{
+    'commands'->{
+        'from <start_pos>'->'set_start',
+        'from <start_pos> to <end_pos>'->'_calculate',
+        'to <end_pos>'->_(end_pos)->(_calculate(null,end_pos)),
+        'toggle_line'->'_toggle_line'
+    },
+    'scope'->'player'
+};
+
+global_position=null;
+global_end_position=null;
+
+global_show_line=false;
+
+global_line = {
+    'start_pos'->global_position,
+    'end_pos'->global_end_position
+};
+
+set_start(pos)->(
+    global_position= pos;
+    print(player(),format('gi Initial point set to: ', 'g '+pos,'?/tp '+(str(pos)-','-'['-']')))
+);
+
+_toggle_line()->(
+    global_show_line= !global_show_line;
+    if(global_show_line,
+        print('Toggled lines on');
+        if(global_position && global_end_position,_draw_line()),//making sure that neither of them are null
+        print('Toggled lines off')
+    );
+    null
+);
+
+_calculate(start_pos, end_pos)->(
+    if(start_pos!=null,
+        set_start(start_pos);
+        calculate(end_pos),
+        if(global_position,
+            calculate(end_pos),
+            print(player(),format('r There was no initial point for '+player()));
+            set_start(end_pos)
```

Comment:
Wait, why do you set start position when given an end one? Should either notify the player you are doing that or not do it at all. _EDIT: The old one does send the notification_

~Also, is there a need say "...initial point **for [player]**"? And in past?~ _EDIT: Nvm found it's parity_

Reply:
I do it cos in the case that that if statement is false, it meant that function received no start pos AND there is no pre-saved start pos, so it would fail if I didnt do that. And I do notify player that im setting the start pos, in the `start_pos()` function at line 23